linux_os/guide/system/network/network-firewalld/ruleset_modifications/ensure_firewall_rules_for_open_ports/oval/shared.xml

```diff
@@ -1,76 +1,300 @@
+      </criteria>
+      <criteria operator="OR">
+        <criterion comment="Check if any service is listening on udp"
+                   test_ref="test_listening_fw_inet_ports_udp_exist" />
+        <criteria operator="AND">
+          <criterion comment="Check there are defined udp service ports in firewall"
+                     test_ref="test_var_firewalled_udp_ports_exists" />
+          <criterion comment="Check firewall rules for udp listening ports"
+                     test_ref="test_listening_fw_inet_service_ports_udp_test" />
+        </criteria>
+        <criteria operator="AND">
+          <criterion comment="Check there are defined udp direct ports in firewall"
+                     test_ref="test_var_firewalled_direct_udp_ports_exists" />
+          <criterion comment="Check firewall rules for directly-defined udp listening ports"
+                     test_ref="test_listening_fw_inet_direct_ports_udp_test" />
+        </criteria>
+      </criteria>
     </criteria>
   </definition>
 
-  <local_variable id="var_firewalled_ports" version="1" comment="Firewalled ports according to firewalld configuration" datatype="int">
-    <object_component object_ref="object_firewalled_service_port" item_field="subexpression"/>
+  <local_variable id="var_firewalled_tcp_ports" version="1" comment="Firewalled ports according to firewalld configuration" datatype="int">
+    <object_component object_ref="object_firewalled_service_tcp_port" item_field="subexpression"/>
+  </local_variable>
+
+  <local_variable id="var_firewalled_direct_tcp_ports" version="1" comment="Directly firewalled ports according to firewalld configuration" datatype="int">
+    <object_component object_ref="object_firewalled_direct_tcp_ports" item_field="subexpression"/>
   </local_variable>
 
 
   <local_variable id="var_firewalled_services" version="1" comment="Firewalld service names" datatype="string">
     <object_component object_ref="object_firewalled_service" item_field="subexpression"/>
   </local_variable>
 
-  <local_variable id="var_all_firewalled_services_desc_filename" version="1" comment="Firewalld service file description" datatype="string">
+  <local_variable id="var_all_firewalled_services_desc_filename" version="1"
+                  comment="Firewalld service file description" datatype="string">
     <concat>
       <variable_component var_ref="var_firewalled_services"/>
       <literal_component>.xml</literal_component>
     </concat>
   </local_variable>
 
+  <ind:variable_object id="object_var_firewalled_tcp_ports" version="1">
+    <ind:var_ref>var_firewalled_tcp_ports</ind:var_ref>
+  </ind:variable_object>
+
+  <ind:variable_object id="object_var_firewalled_direct_tcp_ports" version="1">
+    <ind:var_ref>var_firewalled_direct_tcp_ports</ind:var_ref>
+  </ind:variable_object>
+
+  <linux:inetlisteningservers_state id="state_inet_foreign_port_connected" version="1"
+                                    comment="Checks local address is udp."
```

Comment:
I believe the intention of this state is to locate "listening" ports. This comment is confusing about `udp`.